### DISCUSSION TITLE
MERN app: What is the correct approach when integrating NextAuth with multiple auth providers (e.g. AWS Cognito + Facebook + Google) while still using MongoDB to store the rest of my users' data?

### DISCUSSION THREAD

**Discussion by juanbosnic** [nextauthjs/next-auth#3361] [Help] [answered]
Hello everyone,

I'm building a "MENN" app (Mongo, Express, Nextjs and Nodejs?), and I've decided to use NextAuth to easily integrate multple providers like Facebook, G+ and Amazon Cognito.

However I am not quite sure of a proper way to integrate this authorization solution with the app's own users db.
How do I match users in my app with their corresponding record on the providers's side? Maybe there is a way to send my user's ObjectID to the provider upon registering, so I can match with that?

If this is kind of a noob question it is because I'm just recently cathing up with NextAuth.

Thank you for any help you can give me!

**Comment by ndom91** [comment, marked as answer]
Is this a preexisting application, or a new project you're planning to use NextAuth.js with?

If its a new project, I would recommend using the `users` table that is generated by the library in your adapter of choice. You can then [extend the user model/schema](https://next-auth.js.org/adapters/prisma#create-the-prisma-schema) with any additional fields you may need.

For mongo adapters, we have a `mongoose` based one that is built for mongodb. You could also potentially use the prisma adapter as well if you're looking for an ORM layer on top of mongodb as well, im pretty sure prisma began supporting mongo pretty well recently.

See also:
https://next-auth.js.org/adapters/mongodb
https://next-auth.js.org/adapters/prisma

**Reply by juanbosnic** [reply]
Thank you so much for showing up Nico!

I'm building this app from zero and have full control over its development.
Of course I'll start diving into the solution you proposed.

Cheers!
